app/controllers/answers_controller.rb

```diff
@@ -39,6 +39,7 @@ def destroy
 
   def accept
     return unauthorized_access unless @question.user == current_user
+    return invalid_request('Question has an already accepted answer') if @question.answers.any?(&:accepted)
```

Comment:
do you think a `before_action :authorize_user, :validate_question_already_has_accepted_answer` works in this case?